pkg/chartverifier/pyxis/pyxis.go

```diff
@@ -114,33 +119,51 @@ func IsImageInRegistry(repository string, version string, registry string) (bool
 			json.Unmarshal(body, &registriesBody)
 
 			if len(registriesBody.PyxisRegistries) > 0 {
+
 				var tags []string
+				var shas []string
 				found = false
 				for _, reg := range registriesBody.PyxisRegistries {
-					for _, repo := range reg.Repositories {
-						if repo.Repository == repository && repo.Registry == registry {
-							for _, tag := range repo.Tags {
-								if tag.Name == version {
-									found = true
-									break
+					if len(imageSha) > 0 {
+						if strings.Compare(imageSha, reg.ParsedData.Digest) == 0 {
+							found = true
+						} else {
+							shas = append(shas, reg.ParsedData.Digest)
+						}
+					} else {
+						for _, repo := range reg.Repositories {
+							if repo.Repository == repository && repo.Registry == registry {
+								if len(imageSha) > 0 {
+
 								} else {
-									tags = append(tags, tag.Name)
+									for _, tag := range repo.Tags {
+										if tag.Name == imageTag {
+											found = true
+											break
+										} else {
+											tags = append(tags, tag.Name)
+										}
+									}
 								}
 							}
-						}
-						if found {
-							break
+							if found {
+								break
+							}
 						}
 					}
 					if found {
 						break
 					}
 				}
 				if !found {
-					err = errors.New(fmt.Sprintf("Version %s not found. Found : %s", version, strings.Join(tags, ", ")))
+					if len(imageSha) > 0 {
+						err = errors.New(fmt.Sprintf("Digest %s not found. Found : %s", imageSha, strings.Join(shas, ", ")))
+					} else {
+						err = errors.New(fmt.Sprintf("Tag %s not found. Found : %s", imageTag, strings.Join(tags, ", ")))
+					}
 				}
 			} else {
-				err = errors.New(fmt.Sprintf("Registry not found: %s", registry))
+				err = errors.New(fmt.Sprintf("No images found for Registry/Component: %s/%s", registry, repository))
```

Comment:
do you mean here: `No images found for Registry/Repo` ?

Reply:
Thanks for spotting that, now fixed.